packages/medusa/src/models/payment-session.ts

```diff
@@ -24,12 +16,13 @@ export enum PaymentSessionStatus {
 }
 
 @Unique("OneSelected", ["cart_id", "is_selected"])
+// TODO: This uniq constraint should be updated once the order edit flag is dropped and should add a where clause on cart_id is not null
 @Unique("UniqPaymentSessionCartIdProviderId", ["cart_id", "provider_id"])
 @Entity()
 export class PaymentSession extends BaseEntity {
   @Index()
   @Column({ nullable: true })
-  cart_id: string
+  cart_id: string | null
```

Comment:
question: what context will the cart be null for a payment session?

Reply:
In the case of the payment collection which is composed of resource id but not necessarily for the cart_id 😂 another concept ahah